pkg/operator/hive/hive_controller.go

```diff
@@ -231,6 +243,7 @@ func (r *ReconcileHiveConfig) Reconcile(request reconcile.Request) (reconcile.Re
 			// Object not found, return.  Created objects are automatically garbage collected.
 			// For additional cleanup logic use finalizers.
 			hLog.Debug("HiveConfig not found, deleted?")
+			r.servingCertSecretWatchEstablished = false
```

Comment:
This should only be cleared if the `HiveConfig` deleted was named `hive`. Maybe we should move the name check to before the `Get`.

Reply:
Moved above and changed to not use instance.Name.